### PR TITLE
huds now glitch when they receive EMP damage

### PR DESCRIPTION
**Pull request by Alliostra**
## About The Pull Request

When a hud gets EMP'd, it now displays a glitched, animated overlay. The overlay will go away after 1 minute when hit by light EMP, or 3 minutes when hit by strong EMP.
grafik
and before you ask, no, this won't make EMPs disable NV/thermals/mesons in rigs, I tested it.
## Why It's Good For The Game

Friendly fire by Halicon is now even more funny

## Changelog
:cl:
tweak: huds will now show a glitched overlay when they get EMP'd
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
can you add a way to repair?

File: code/modules/clothing/glasses/glasses.dm [hunk 1]
```diff
@@ -55,3 +55,8 @@
 	if(istype(Z,/obj/item/clothing/glasses/attachable_lenses))
 		var/obj/item/clothing/glasses/attachable_lenses/lenses = Z
 		lenses.handle_insertion(src, user)
+
+/obj/item/clothing/glasses/emp_act(severity)
+	. = ..()
+	if(hud)
+		hud.emp_act(severity)
```
Comment (TheShown911) [inline code/modules/clothing/glasses/glasses.dm:62]: Newline missing.

File: code/modules/clothing/glasses/hud.dm [hunk 2]
```diff
@@ -8,14 +8,28 @@
...
+/obj/item/clothing/glasses/hud/emp_act(severity)
+	. = ..()
+	malfunctioning = TRUE
+	switch(severity)
+		if(1)
+			spawn(1 MINUTES)
+			malfunctioning = FALSE
+		if(2)
+			spawn(3 MINUTES)
+			malfunctioning = FALSE
```
Comment (TheShown911) [inline code/modules/clothing/glasses/hud.dm:31]: Spawn is cringe, use callback timers.

**Review by TheShown911 — APPROVED** [review]

**Review by hyperioo — DISMISSED** [review]
See Shown's comments

File: code/modules/clothing/glasses/glasses.dm (hunk 1 above)
Comment (Alliostra) [reply code/modules/clothing/glasses/glasses.dm]: Addressed

File: code/modules/clothing/glasses/hud.dm (hunk 2 above)
Comment (Alliostra) [reply code/modules/clothing/glasses/hud.dm]: Addressed

**Pull request merged**